### PR TITLE
Add Postgres Json functions: `JSON_BUILD_OBJECT` and `JSON_AGG`

### PR DESCRIPTION
**Pull request by IgnisDa**
## PR Info

This PR adds the `JSON_BUILD_OBJECT` and `JSON_AGG` functions along with docs and tests.

## Additional

Also addressed a few clippy lints.

### CODE REVIEW

**Comment by IgnisDa** [comment]
@billy1624 Can you please merge this?

**Comment by IgnisDa** [comment]
@tyt2y3 Wondering if you could take a look at this PR?

**Comment by IgnisDa** [comment]
@tyt2y3 Could you please take a look at this?

File: src/expr.rs
Comment (tyt2y3) [inline src/expr.rs:1]: thank you for fixing this as well!

**Review by tyt2y3 — APPROVED** [review]

**Comment by tyt2y3** [comment]
Thank you!

**Pull request merged**